backend/src/main/java/com/redhat/cloud/notifications/db/repositories/EndpointRepository.java

```diff
@@ -389,4 +389,19 @@ public Optional<Endpoint> getEndpointWithLinkedEventTypes(String orgId, UUID id)
             return Optional.empty();
         }
     }
+
+    /**
+     * Retrieves a list of endpoints.
+     * @param limit the size of the list to retrieve.
+     * @param offset the offset we should apply to skip the already read
+     *               endpoints.
+     * @return a list of endpoints.
+     */
+    public List<Endpoint> getEndpointsWithLimitAndOffset(final int limit, final int offset) {
+        return this.entityManager
+            .createQuery("FROM Endpoint", Endpoint.class)
```

Comment:
I think we shoudl remove system endpoints ( when orgId is null)

Reply:
You are right, yes. I updated the query!